Xamarin.Forms.Platform.Android/AppCompat/NavigationPageRenderer.cs

```diff
@@ -173,13 +152,52 @@ protected override void Dispose(bool disposing)
 				if (_drawerLayout != null && _drawerListener != null)
 				{
 					_drawerLayout.RemoveDrawerListener(_drawerListener);
+				}
+
+				if (_drawerListener != null)
```

Comment:
This can be moved to its own if block since the previous `_drawerLayout != null && _drawerListener != null` may evaluate to false.